yb-voyager/cmd/importData.go

```diff
@@ -935,34 +935,67 @@ func executeSqlFile(file string) {
 	}
 
 	var errOccured = 0
-	sqlStrArray := createSqlStrArray(file, "")
-	for _, sqlStr := range sqlStrArray {
-		log.Infof("Run query %q on target %q", sqlStr[1], target.Host)
-		_, err := conn.Exec(context.Background(), sqlStr[0])
+	sqlInfoArr := createSqlStrInfoArray(file, objType)
+	for _, sqlInfo := range sqlInfoArr {
+		err := executeSqlStmtWithRetries(conn, sqlInfo, objType)
 		if err != nil {
-			log.Errorf("Run query %q on target %q: %s", sqlStr[1], target.Host, err)
-			if strings.Contains(err.Error(), "already exists") {
+			errOccured = 1
+		}
+	}
+
+	utils.WaitChannel <- errOccured
+	<-utils.WaitChannel
+}
+
+func executeSqlStmtWithRetries(conn *pgx.Conn, sqlInfo sqlInfo, objType string) error {
+	var err error
+	retryCount := 0
+	log.Infof("Run query %q on target %q", sqlInfo.formattedStmtStr, target.Host)
+	for retryCount <= DDL_MAX_RETRY_COUNT {
+		_, err = conn.Exec(context.Background(), sqlInfo.stmt)
```

Comment:
I am creating a new connection in case of that error:
```
if strings.Contains(strings.ToLower(err.Error()), strings.ToLower(SCHEMA_VERSION_MISMATCH_ERR)) &&
objType == "INDEX" {
       // creating fresh connection
       conn, err2 := pgx.Connect(context.Background(), target.GetConnectionUri())
}
```

Reply:
done